web-console/src/utils/sampler.ts

```diff
@@ -202,7 +203,10 @@ function makeSamplerIoConfig(
     ioConfig = deepSet(ioConfig, 'useEarliestSequenceNumber', sampleStrategy === 'start');
   }
   // In order to prevent potential data loss null columns should be kept by the sampler and shown in the ingestion flow
-  ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);
+  const reingestMode = isDruidSourceFromInputSource(deepGet(ioConfig, 'inputSource'));
+  if (!reingestMode) {
+    ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);
```

Comment:
the issue is that `ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);` creates the `ioConfig.inputFormat` object even if it is not there. Right?
I think the fix is to change:

```
ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);
```

to:

```
if (ioConfig.inputFormat) {
  ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);
}
```

And not worry about the `reingestMode` here (we want to set `keepNullColumns` even if we are not re-ingesting)

Reply:
Done